src/main/scala/shine/cuda/BuiltInAttribute.scala

```diff
@@ -0,0 +1,48 @@
+package shine.cuda
+
+import arithexpr.arithmetic.{
+  ArithExprFunctionCall, Range, RangeUnknown, Sign, SimplifiedExpr}
+import shine.DPIA.Nat
+import shine.OpenCL.BuiltInFunctionCall
+
+class BuiltInAttribute private(name: String, val param: Char, range: Range)
```

Comment:
was is this class modeling exactly? maybe a comment would help.

Reply:
This model the blockdim and threadId attributes in CUDA similar to the getLocalSize() and getGlobalSize() functions (BuiltInFunctionCall) in OpenCL.